daft/dataframe/dataframe.py

```diff
@@ -214,11 +214,15 @@ def __iter__(self) -> Iterator[Dict[str, Any]]:
                     yield row
 
     @DataframePublicAPI
-    def iter_partitions(self) -> Iterator[Union[MicroPartition, "RayObjectRef"]]:
+    def iter_partitions(self, buffer_size: Optional[int] = None) -> Iterator[Union[MicroPartition, "RayObjectRef"]]:
```

Comment:
From the looks of it, `iter_partitions` seems like a public API. In that case, do we want to enforce validations on the buffer_size to be > 0?

Reply:
Yes, I actually realized that this was missing from our documentation. This is a public API.

Good catch, it will need to be `> 1` here. Adding validations!